main/test/ca/uwaterloo/flix/library/TestStringBuilder.flix

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright 2019 Stephen Tetley
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/////////////////////////////////////////////////////////////////////////////
+// new                                                                     //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def new01(): StringBuilder = StringBuilder.new()
+
+/////////////////////////////////////////////////////////////////////////////
+// append                                                                  //
+/////////////////////////////////////////////////////////////////////////////
+@test 
+def append01(): Bool = 
+    let sb1 = StringBuilder.new();
+    let sb2 = StringBuilder.new();
+    StringBuilder.append(sb1, sb2);
+    StringBuilder.toString(sb1) `assertEq!` ""

```

Comment:
I *think* we are moving away from using `assert!` and `assertEq`, instead a test case should simply return a true boolean if it was successful.

(I know this is not documented anywhere, and many tests still use it. The reasoning is that `assertEq` throws an exception, but we do not plan to have actual exceptions, so it is somewhat "magical" and I want to move away from that).